packages/markup-include/src/components/EditorPanel.tsx

```diff
@@ -0,0 +1,526 @@
+                : conditionLookup[columnName].map(valueItem => {
+                    return (
+                      <option key={valueItem} value={valueItem}>
+                        {valueItem}
+                      </option>
+                    )
+                  })}
+            </select>
+          </div>
+        </label>
+      </div>
+    </>
+  )
+}
+
+type Variable = {
+  columnName: string
+  conditions: Condition[]
+  name: string
+  tag: string
+}
+
+type Condition = {
+  columnName: string
+  isOrIsNotEqualTo: string
+  value: string
+}
+
+type VariableSectionProps = {
+  allVariableNames: string[]
+  controls: OpenControls
+  data: Object[]
+  deleteVariable: Function
+  updateVariableArray: Function
+  variableConfig: Variable
+  variableIndex: number
+}
+
+const VariableSection: React.FC<VariableSectionProps> = ({ allVariableNames, controls, data, deleteVariable, updateVariableArray, variableConfig, variableIndex }) => {
+  const [openVariableControls, setOpenVariableControls] = controls
+  const show = openVariableControls[variableIndex]
+  const setShow = (key, value) => {
+    setOpenVariableControls({ ..._.cloneDeep(openVariableControls), [key]: value })
+  }
+
+  const openConditionControls = useState({})
+
+  const columnNames = Object.keys(data[0])
+  const [selectedColumn, setNewVariableColumnName] = useState(variableConfig.columnName)
+  const [conditionsList, setConditionsList] = useState(variableConfig.conditions)
+  const [variableName, setVariableName] = useState(variableConfig.name)
+
+  const conditionLookup: Record<string, string[] | number[]> = useMemo(() => {
+    return columnNames.reduce((acc, column) => {
+      acc[column] = _.uniq(data.map(row => row[column]))
+      return acc
+    }, {})
+  }, [columnNames])
+
+  const handleVariableColumnChange = (columnName: string) => {
+    setNewVariableColumnName(columnName)
+    setConditionsList([])
+  }
+
+  const updateConditionsList = (conditionSettings: Condition, number: number) => {
+    const { columnName, isOrIsNotEqualTo, value } = conditionSettings
+    const newConditionsList = [..._.cloneDeep(conditionsList)]
+    newConditionsList[number] = {
+      columnName: columnName,
+      isOrIsNotEqualTo: isOrIsNotEqualTo,
+      value: value
+    }
+    setConditionsList(newConditionsList)
+  }
+
+  const removeCondition = (conditionNumber: number) => {
+    const updatedConditionsList = [..._.cloneDeep(conditionsList)]
+    updatedConditionsList.splice(conditionNumber, 1)
+    setConditionsList(updatedConditionsList)
+  }
+
+  const handleAddConditionClick = () => {
+    const newConditionsList = [..._.cloneDeep(conditionsList)]
+    newConditionsList.push({
+      columnName: '',
+      isOrIsNotEqualTo: 'is',
+      value: ''
+    })
+    setConditionsList(newConditionsList)
+    const [conditionControls, setConditionControls] = openConditionControls
+    setConditionControls({ ..._.cloneDeep(conditionControls), [conditionsList.length + 1]: true })
```

Comment:
don't deep clone here

Reply:
Also it looks like you're using a Record as though it were an array?

Reply:
Removed deep clone but, need to talk through the Record as an array.